rest_framework/fields.py

```diff
@@ -71,7 +71,11 @@ def get_attribute(instance, attrs):
         except ObjectDoesNotExist:
             return None
         if is_simple_callable(instance):
-            instance = instance()
+            try:
+                instance = instance()
+            except (AttributeError, KeyError) as exc:
```

Comment:
Let's also add a comment here. Something like...

    # If we raised an Attribute or KeyError here it'd get treated as an omitted field
    # in `Field.get_attribute()`. Instead we raise a ValueError to ensure the exception
    # is not masked.